src/modm/platform/eth/stm32/eth.hpp

```diff
@@ -0,0 +1,705 @@
+		Pbl5                  = modm::Bit13,
+		Pbl4                  = modm::Bit12,
+		Pbl3                  = modm::Bit11,
+		Pbl2                  = modm::Bit10,
+		Pbl1                  = modm::Bit9,
+		Pbl0                  = modm::Bit8,
+		EnhancedDescFormat    = modm::Bit7,
+		DescriptorSkipLength4 = modm::Bit6,
+		DescriptorSkipLength3 = modm::Bit5,
+		DescriptorSkipLength2 = modm::Bit4,
+		DescriptorSkipLength1 = modm::Bit3,
+		DescriptorSkipLength0 = modm::Bit2,
+		DmaArbitration        = modm::Bit1,
+		SoftwareReset         = modm::Bit0,
+	};
+	MODM_FLAGS32(DmaBusMode);
+
+	enum class
+	BurstLength : uint32_t {
+		Length1Beat   = 0b000001,
+		Length2Beats  = 0b000010,
+		Length4Beats  = 0b000100,
+		Length8Beats  = 0b001000,
+		Length16Beats = 0b010000,
+		Length32Beats = 0b100000,
+	};
+	using BurstLength_t = modm::Configuration<DmaBusMode_t, BurstLength, 0b111111, 8>;
+	using RxDmaBurstLength_t = modm::Configuration<DmaBusMode_t, BurstLength, 0b111111, 17>;
+
+	template <Peripheral peripheral, template <Peripheral _> class... Signals>
+	struct GpioConfigurator;
+	template <Peripheral peripheral, template <Peripheral _> class Signal, template <Peripheral _> class... Signals>
+	struct GpioConfigurator<peripheral, Signal, Signals...>
+	{
+		static inline void
+		configure() {
+			Signal<peripheral>::Gpio::configure(modm::platform::Gpio::OutputType::PushPull, modm::platform::Gpio::OutputSpeed::VeryHigh);
+			GpioConfigurator<peripheral, Signals...>::configure();
+		}
+	};
+	template <Peripheral peripheral>
+	struct GpioConfigurator<peripheral>
+	{
+		static inline void
+		configure() {}
+	};
+
+public:
+	template <template<Peripheral _> class... Signals>
+	static void
+	connect()
+	{
+		using Configurator = GpioConfigurator<Peripheral::Eth, Signals...>;
+		using Connector = GpioConnector<Peripheral::Eth, Signals...>;
+
+//		Ethernet::RMII_Ref_Clk::configure(modm::platform::Gpio::OutputType::PushPull, modm::platform::Gpio::OutputSpeed::VeryHigh);
+//		Ethernet::RMII_Mdc::configure(modm::platform::Gpio::OutputType::PushPull, modm::platform::Gpio::OutputSpeed::VeryHigh);
+//		Ethernet::RMII_Mdio::configure(modm::platform::Gpio::OutputType::PushPull, modm::platform::Gpio::OutputSpeed::VeryHigh);
+//		Ethernet::RMII_Crs_Dv::configure(modm::platform::Gpio::OutputType::PushPull, modm::platform::Gpio::OutputSpeed::VeryHigh);
+//		Ethernet::RMII_Tx_En::configure(modm::platform::Gpio::OutputType::PushPull, modm::platform::Gpio::OutputSpeed::VeryHigh);
+//		Ethernet::RMII_Tx_D0::configure(modm::platform::Gpio::OutputType::PushPull, modm::platform::Gpio::OutputSpeed::VeryHigh);
+//		Ethernet::RMII_Tx_D1::configure(modm::platform::Gpio::OutputType::PushPull, modm::platform::Gpio::OutputSpeed::VeryHigh);
+//		Ethernet::RMII_Rx_D0::configure(modm::platform::Gpio::OutputType::PushPull, modm::platform::Gpio::OutputSpeed::VeryHigh);
+//		Ethernet::RMII_Rx_D1::configure(modm::platform::Gpio::OutputType::PushPull, modm::platform::Gpio::OutputSpeed::VeryHigh);
```

Comment:
Why is this code commented out?
Are the GPIOs initialized somewhere else?

Reply:
Yes, this is done by the configurator and connector (see line 382). Need to remove the lines.